src/lib.rs

```diff
@@ -56,9 +55,11 @@ impl CallbackThreadPool {
         for _ in 0..thread_num {
             let receiver = cb_receiver.clone();
             thread_pool.push(thread::spawn(move || {
+                // Create a dedicated runtime for the thread
+                let rt = tokio::runtime::Runtime::new().expect("Unable to create runtime");
```

Comment:
I suppose this was the case implicitly earlier using async-std as well, seems like we're starting a new tokio runtime per thread.

I haven't used tokio much -- is this heavy? Standard practice when wanting a dedicated async thread?

Reply:
IMHO, if we use the thread pool to handle the callback, I don't think we need async anymore. However, the callback is async func, so it forces us to create a runtime to run the async func.
In fact, I'm thinking about whether using thread pool here is a good idea or not. Maybe we should have a dedicated runtime here (with multithreaded config) instead of running async func in every single thread.

Reply:
> Maybe we should have a dedicated runtime here (with multithreaded config) instead of running async func in every single thread.

:thinking: Honestly, I dunno which is more performant or the trade-offs.

Maybe list up the trade-offs and then do some relatively quick testing?